src/test/scala/zio/s3/S3Test.scala

```diff
@@ -53,25 +54,46 @@ object S3Suite {
       testM("list objects") {
         for {
           succeed <- listObjects_(bucketName)
-        } yield assert(succeed.bucketName)(equalTo(bucketName)) && assert(succeed.objectSummaries)(
+        } yield assert(succeed.bucketName)(equalTo(bucketName)) && assert(
+          succeed.objectSummaries.map(noModificationTimeSummary)
+        )(
           hasSameElements(
             List(
-              S3ObjectSummary(bucketName, "console.log"),
-              S3ObjectSummary(bucketName, "dir1/hello.txt"),
-              S3ObjectSummary(bucketName, "dir1/user.csv")
+              S3ObjectSummary(bucketName, "console.log", FakeModificationTime, 33),
+              S3ObjectSummary(bucketName, "dir1/hello.txt", FakeModificationTime, 36),
+              S3ObjectSummary(bucketName, "dir1/user.csv", FakeModificationTime, 181)
             )
           )
         )
       },
       testM("list objects with prefix") {
         for {
           succeed <- listObjects(bucketName, "console", 10)
-        } yield assert(succeed)(
+        } yield assert(succeed.copy(objectSummaries = succeed.objectSummaries.map(noModificationTimeSummary)))(
```

Comment:
i would prefer this approach

```scala

assert(succeed)(
          hasField("bucketName", (l: S3ObjectListing) => l.bucketName, equalTo(bucketName)) &&
          hasField("objectSummaries",
              (l: S3ObjectListing) => l.objectSummaries.map(o => o.bucketName -> o.key),
              equalTo(Chunk.single((bucketName, "console.log")))
          )
        )
```

Reply:
I thought about this approach. Why I decided to avoid it is because the `S3ObjectListing` can be extended even further, for example, with the `eTag` field. When using the `hasField` the compilation of the test won't fail. With the current approach the person who adds the additional field is enforced to adjust the test, because it requires all the field to instantiate the case class in the test case.